src/app/reporting/LoggingManagement.h

```diff
@@ -0,0 +1,249 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    Copyright (c) 2015-2017 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ *
+ * @brief
+ *   Management of the CHIP Event Logging.
+ *
+ */
+#pragma once
+
+#ifndef _CHIP_DATA_MODEL_EVENT_LOGGING_MANAGEMENT_H
+#define _CHIP_DATA_MODEL_EVENT_LOGGING_MANAGEMENT_H
+
+#include "EventLoggingTypes.h"
+#include <core/CHIPCircularTLVBuffer.h>
+#include <messaging/ExchangeMgr.h>
+#include <support/PersistedCounter.h>
+#include <util/basic-types.h>
+
+#define CHIP_CONFIG_EVENT_SIZE_RESERVE 196
+
+namespace chip {
+namespace app {
+namespace reporting {
+
+/**
+ * @brief
+ *   Internal event buffer, built around the chip::TLV::CHIPCircularTLVBuffer
+ */
+
+class CircularEventBuffer : public TLV::CHIPCircularTLVBuffer
+{
+public:
+    // for doxygen, see the CPP file
+    CircularEventBuffer(uint8_t * apBuffer, uint32_t aBufferLength, CircularEventBuffer * apPrev, CircularEventBuffer * apNext);
+
+    CHIP_ERROR GetNextBuffer(chip::TLV::TLVReader & aReader, const uint8_t *& aBufStart, uint32_t & aBufLen) override;
+
+    // for doxygen, see the CPP file
+    bool IsFinalDestinationForPriority(PriorityLevel aPriority) const;
+
+    chip::EventId VendEventID(void);
+    void RemoveEvent(uint16_t aNumEvents);
+
+    // for doxygen, see the CPP file
+    void AddEvent(timestamp_t aEventTimestamp);
+
+    CircularEventBuffer * mpPrev; ///< A pointer #CircularEventBuffer storing events less important events
```

Comment:
could we avoid public data members in classes?

Reply:
agree, thanks!